nbbuild/antsrc/org/netbeans/nbbuild/ParseProjectXml.java

```diff
@@ -991,7 +993,16 @@ private String computeClasspath(String myCNB, ModuleListParser modules, Dep[] de
                 if (!dep.impl && /* #71807 */ dep.run && !recursive && !runtime) {
                     String friends = attr.getValue("OpenIDE-Module-Friends");
                     if (friends != null && !Arrays.asList(friends.split(" *, *")).contains(myCNB)) {
-                        throw new BuildException("The module " + myCNB + " is not a friend of " + depJar, getLocation());
+                        String friendIssue = "The module " + myCNB + " is not a friend of " + depJar;
+                        if (Boolean.parseBoolean(getProject().getProperty(PROP_FRIENDLY_MODULES))) {
+                            log(friendIssue, Project.MSG_WARN);
+                        } else {                           
+                            log("The " + depJar.getName() + " is a non-stable API/SPI. It is only available for its friends.", Project.MSG_ERR);
```

Comment:
Shouldn't the right fix be to turn the non-stable API/SPI into stable one? NetBeans was using friend dependencies too much. Probably my mistake; maybe the switch to Apache is the right time to change the attitude.

Reply:
+10 (OK, 1!) to that.  And external libs, too, please! :smile:  Friend makes sense to me for splitting out functionality, but maybe we need something else entirely for non-stable?

Reply:
Well I think the module versioning, stable/non-stable/friendliness of the API-s needs a redesign/rethink. However I think that is more feasible after the Apache migration has happened. Right now there is too much going on on the entire codebase.
Though if somebody would like to announce a few API/SPI as stable, I just could only welcome that. I even have a few candidates for that as well...

Reply:
I suspect that it would be difficult to back out from the behavior proposed here if it was added. So, it would seem better to me to not do changes like this unless we are sure about them.

I guess it might be useful to know the views/proposals on the redesign/rethink of the versioning, etc. Not sure if the concepts really need a redesign (although the use of the concepts may sometimes need more "strictness"), but I guess I'd suggest to start to discuss that if you think that's what's needed.

In any case, I guess it might be good to discuss modules which are causing the problems.

One thing I'd like to point out: it is possible to force compilation against any module by putting its path (path to the module jar) to cp.extra (and changing the dependency to runtime only).

Reply:
Well, the proposed behavior is already available for those who are using Maven to build plugins. (Though I prefer the Ant harness as it has better support.)

Just a reading for rethink: http://wiki.netbeans.org/NbmPackageStability#Usage_from_external_modules
It is written by Jarda and JGlick and a good point to start. Though I still feel this shall be post 9.0